Added discussion regarding the refactoring

Samuelhertzberg opened the pull request:
Added pros, cons and summary to illustrate and discuss the refactoring made, should resolve #5

Samuelhertzberg committed:
Added discussion regarding the refactoring

Pros
cons
summary.
#5

francislaus reviewed (changes requested):
I really like your text - it is well written.

But I think that 'Benefits' should start with a lowercase character.

Jonagoo committed:
Fixed uppercase benefits

Benefits was written with an uppercase b in the middle of a sentence. This should resolve #5

francislaus reviewed (approved):
Nice!
